BuildingBlocks/src/UnitTestTools/FluentValidation/Extensions.cs

```diff
@@ -36,12 +36,30 @@ public static IEnumerable<ValidationFailure> ShouldHaveValidationErrorForItemWit
 
 public static class ValidationTestExtensions
 {
-    public static void ShouldHaveValidationErrorForItem<T>(this TestValidationResult<T> testValidationResult, string propertyName, string expectedErrorCode,
-        string expectedErrorMessage)
+    public static void ShouldHaveValidationErrorForItem<T>(this TestValidationResult<T> testValidationResult, string propertyName, string expectedErrorCode, string expectedErrorMessage)
     {
-        var validationError = testValidationResult.Errors.FirstOrDefault(r => r.PropertyName.Contains(propertyName)) ?? throw new XunitException($"Expected error for property '{propertyName}'.");
+        var errorsForProperty = testValidationResult.ShouldHaveValidationErrorFor(propertyName);
+        errorsForProperty.Should().Contain(r => r.ErrorCode == expectedErrorCode && r.ErrorMessage == expectedErrorMessage);
+    }
+
+    public static void ShouldHaveValidationErrorForItemInCollection<T>(this TestValidationResult<T> testValidationResult, string collectionWithInvalidId, int indexWithInvalidId,
```

Comment:
This method isn't used, as far as I can see

Reply:
Thought it might add some value having an extension method such as this at the ready, considering the other 3 existing ones.
But I'll remove it if you think it doesn't make sense to have this in advance.